lib/Dialect/FIRRTL/FIRRTLFolds.cpp

```diff
@@ -55,6 +55,17 @@ static Value moveNameHint(OpResult old, Value passthrough) {
   return passthrough;
 }
 
+template <typename BaseTy>
+static Value subfieldOrIndex(PatternRewriter &rewriter, Location loc,
+                             BaseTy base, IntegerAttr index) {
+  return TypeSwitch<Type, Value>(base.getType())
+      .template Case<FVectorType>(
+          [&](auto _) { return rewriter.create<SubindexOp>(loc, base, index); })
+      .template Case<BundleType>([&](auto _) {
+        return rewriter.create<SubfieldOp>(loc, base, index);
```

Comment:
MSVC sometimes has problems with `template` inside `TypeSwitch`. The `template` can be avoided by avoiding `auto`, e.g., `[&](FVectorType _){ /* ... */ };`.

Reply:
I swear I tried that, but maybe not :sweat_smile:, thanks!  In the past I thought being explicit about the type `Case<FVectorType>` was the trick, is it just the lambda, or both (do you know)?

Anyway I'll switch to specifying the type in both the `Case<T>` and in the lambda, thanks!

Reply:
Both are needed as far as I know. It's more verbose, but I know MSVC is happy about it (and I subjectively don't like having to sprinkle `template` around 😉 ).

Reply:
(Dropped this function entirely)

Reply:
> Both are needed as far as I know. It's more verbose, but I know MSVC is happy about it

Awesome, good to know.  Thanks!

> and I subjectively don't like having to sprinkle `template` around wink

:100: